rlp/raw_test.go

```diff
@@ -283,3 +283,32 @@ func TestAppendUint64Random(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestBytesSize(t *testing.T) {
+	tests := []struct {
+		v    []byte
+		size uint64
+	}{
+		{v: []byte{}, size: 1},
+		{v: []byte{0x1}, size: 1},
+		{v: []byte{0x7E}, size: 1},
+		{v: []byte{0x7F}, size: 2},
+		{v: []byte{0x80}, size: 2},
+		{v: []byte{0xFF}, size: 2},
+		{v: []byte{0xFF, 0xF0}, size: 3},
+		{v: make([]byte, 55), size: 56},
+		{v: make([]byte, 56), size: 58},
+	}
+
+	for _, test := range tests {
+		s := BytesSize(test.v)
+		if s != test.size {
+			t.Errorf("BytesSize(%#x) -> %d, want %d", test.v, s, test.size)
+		}
```

Comment:
Wouldn't it be nicer (more easily extendable) to do something like this instead? Why hardcode... ? 
```suggestion
		have := BytesSize(test.v)
		enc, _ := EncodeToBytes(test.v)
		want := len(enc)
		if have != want {
			t.Errorf("BytesSize(%#x) -> %d, want %d", test.v, have, want)
		}
```

Reply:
Good call! I've added an additional check to compare with the length returned by EncodeToBytes here, and this found a bug in BytesSize!